src/index.js

```diff
@@ -148,11 +160,11 @@ IdTokenVerifier.prototype.verify = function(token, requestedNonce, cb) {
       );
     }
 
-    if (_this.issuer !== iss) {
+    if ((typeof _this.issuer == 'function' && _this.issuer(jwt.payload) !== iss) || (typeof _this.issuer != 'function' && _this.issuer !== iss)) {
```

Comment:
> _this.issuer(jwt.payload)

Does it make sense to cache this value? We should avoid calling `_this.issuer` twice in the event of an error.